internal/provider_test/integration_test.go

```diff
@@ -1411,5 +1428,74 @@ func TestValidation(t *testing.T) {
 		require.Error(t, err)
 
 		require.Contains(t, string(out), errorString)
+		require.Contains(t, string(out), errorDescription)
+
+		vars.providerCfg.Password = nil
+
+		// And passing password via the environment
+		t.Setenv(NUODB_CP_PASSWORD, "password")
+
+		tf.WriteConfigT(t, vars.builder.Build())
+
+		// Run `terraform validate`
+		out, err = tf.Validate()
+		require.Error(t, err)
+
+		require.Contains(t, string(out), errorString)
+		require.Contains(t, string(out), errorDescription)
+
+		t.Setenv(NUODB_CP_PASSWORD, "")
```

Comment:
Why is this necessary? I though `testing.T.Setenv()` restored the original setting? And we don't know what the original setting was anyway, so it would not make sense to just set it to empty.

Reply:
It is not and I removed it.